packages/jaeger-ui/src/components/Monitoring/ServicesView/serviceGraph.tsx

```diff
@@ -0,0 +1,221 @@
+  }
+
+  renderLines() {
+    const { metricsData, color } = this.props;
+
+    if (metricsData) {
+      const graphs: any = [];
+      let i = 0;
+
+      this.getData().forEach((line: ServiceMetricsObject, idx: number) => {
+        graphs.push(
+          <AreaSeries
+            key={i++}
+            data={line.metricPoints ? line.metricPoints : []}
+            getNull={(d: Points) => d.y !== null}
+            onNearestX={
+              idx === 0
+                ? (_datapoint: Points, { index }: { index: number }) => {
+                    this.setState({
+                      crosshairValues: this.getData().map((d: ServiceMetricsObject) => ({
+                        ...d.metricPoints[index],
+                        label: d.quantile,
+                      })),
+                    });
+                  }
+                : null
+            }
+            opacity={0.1}
+            color={[color || this.colors[idx]]}
+          />
+        );
+        graphs.push(
+          <LineSeries
+            getNull={(d: Points) => d.y !== null}
+            key={i++}
+            data={line.metricPoints ? line.metricPoints : []}
+            color={[color || this.colors[idx]]}
+          />
+        );
+      });
+
+      return graphs;
+    }
+
+    return [];
+  }
+
+  generatePlaceholder(placeHolder: string | JSX.Element) {
+    const { width } = this.props;
+
+    return (
+      <div
+        className="center-placeholder"
+        style={{
+          width,
+          height: this.height - 74,
+        }}
+      >
+        {placeHolder}
+      </div>
+    );
+  }
+
+  render() {
+    const {
+      width,
+      yDomain,
+      showHorizontalLines,
+      showLegend,
+      loading,
+      metricsData,
+      marginClassName,
+      name,
+      error,
+    } = this.props;
+    let GraphComponent = this.generatePlaceholder(<LoadingIndicator centered />);
+    const noDataComponent = this.generatePlaceholder('No Data');
+    const apiErrorComponent = this.generatePlaceholder('Couldn’t fetch data');
+
+    const Plot = (
+      <XYPlot
+        margin={{ bottom: 25 }}
+        onMouseLeave={() => this.setState({ crosshairValues: [] })}
+        width={width}
+        height={this.height - 74}
+        yDomain={yDomain}
+      >
+        {showHorizontalLines ? <HorizontalGridLines /> : null}
+        <XAxis tickFormat={tickFormat} tickTotal={Math.floor(width / 60)} />
+        <YAxis />
+        {this.renderLines()}
+        <Crosshair values={this.state.crosshairValues}>
+          <div style={{ width: 140 }}>
+            {this.state.crosshairValues[0] &&
+              `${new Date(this.state.crosshairValues[0].x).toLocaleDateString()} ${new Date(
+                this.state.crosshairValues[0].x
+              ).toLocaleTimeString()}`}
+            {this.state.crosshairValues.map((d: TCrossHairValues) =>
+              showLegend ? (
+                <div key={d.label}>
+                  {d.label * 100}th: {d.y}
```

Comment:
I feel using the P-notation is more correct as it explicitly communicates that it's percentile-based rather than other quantile types.

Also, though highly unlikely, we won't have to worry about handling `st`, `nd` and `rd` suffixes.

```suggestion
                  P{d.label * 100}: {d.y}
```